src/test/test-http.c

```diff
@@ -20,6 +20,93 @@
 #include "sol-util-internal.h"
 #include "sol-http.h"
 #include "sol-vector.h"
+#include "sol-util.h"
+
+DEFINE_TEST(go);
+
+static void
+go(void)
+{
+    size_t i;
+    int r;
+
+    static const struct {
+        const struct sol_str_slice content_type;
+        double qvalue;
+        size_t tokens_size;
+        const char *tokens[10];
+    } map [][100] = {
+        { { SOL_STR_SLICE_LITERAL("text/html"), 1.0, 0, { 0 } } },
+        {
+            { SOL_STR_SLICE_LITERAL("audio/basic"), 1.0, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("audio/*"), 0.2, 0, { 0 } }
+        },
+        {
+            { SOL_STR_SLICE_LITERAL("text/html"), 1.0, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("text/x-c"), 1.0, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("text/x-dvi"), 0.8, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("text/plain"), 0.5, 0, { 0 } },
+        },
+        {
+            { SOL_STR_SLICE_LITERAL("text/html"), 1.0, 1, { "level=1" } },
+            { SOL_STR_SLICE_LITERAL("text/html"), 1.0, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("text/*"), 1.0, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("*/*"), 1.0, 0, { 0 } },
+        },
+        {
+            { SOL_STR_SLICE_LITERAL("text/html"), 1.0, 1, { "level=1" } },
+            { SOL_STR_SLICE_LITERAL("text/html"), 0.7, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("text/html"), 0.4, 2, { "level=2", "level=3" } },
+            { SOL_STR_SLICE_LITERAL("text/*"), 0.3, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("*/*"), 0.5, 0, { 0 } },
+        },
+        {
+            { SOL_STR_SLICE_LITERAL("text/html"), 1.0, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("text/*"), 0.8, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("image/gif"), 0.6, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("image/jpeg"), 0.6, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("image/*"), 0.5, 0, { 0 } },
+            { SOL_STR_SLICE_LITERAL("*/*"), 0.1, 0, { 0 } },
+        },
+    };
+
+    static const struct {
+        int r;
+        size_t priorities_len;
+        struct sol_str_slice accept;
+    } test[] = {
```

Comment:
ugh, can't you add the expected results here as well? This lookup will likely introduce errors in the future.